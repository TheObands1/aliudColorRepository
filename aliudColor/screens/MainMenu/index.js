import React, { useState, useEffect}  from 'react';
import { View, Text, Image, TouchableOpacity } from "react-native";
import MainMenuStyle from './styles'
import { GameTitle } from '../../components';

export default function MainMenuView() {

  const [isSoundOn, setIsSoundOn] = useState(true);

  function onPressPlayButton() {
    console.log("onPressPlayButton event handler");
  };

  function onPressLeaderboardButton() {
    console.log("onPressLeaderboardButton event handler");
  };

  function changeSoundIcon(){
    if(isSoundOn)
    {
       return require("../../assets/icons/speaker-on.png")
    }
    else{
      return require("../../assets/icons/speaker-off.png")
    }

  }

  function onToggleSound() {
    if(isSoundOn)
    {
      setIsSoundOn(false);
      console.log("onToggleSound is false");
    }
    else{
      setIsSoundOn(true);
      console.log("onToggleSound is true");
    }

  };


  return (<View style={MainMenuStyle.generalStyle}>
            <GameTitle/>

            {/*Should this ↓ touchableOpacity be its own component? */}
            <TouchableOpacity onPress={onPressPlayButton} style={MainMenuStyle.playButtonArea}>
              <Image source={require("../../assets/icons/play_arrow.png")} style={MainMenuStyle.playIcon}/>
              <Text style={MainMenuStyle.playText}>play!</Text>
            </TouchableOpacity>
             {/*Should this ↑ touchableOpacity be its own component? */}

            {/*Should this ↓ text part be its own component? */}
            <View style={MainMenuStyle.highScoreArea}>
              <Image source={require("../../assets/icons/trophy.png")} style={MainMenuStyle.highScoreIcon}/>
              <Text style={MainMenuStyle.highScoreText}>high-score: 0</Text>
            </View>
             {/*Should this ↑ text part be its own component? */} 

             {/*Should this ↓ touchableOpacity be its own component? */}
            <TouchableOpacity onPress={onPressLeaderboardButton} style={MainMenuStyle.leaderboardArea}>
              <Image source={require("../../assets/icons/leaderboard.png")} style={MainMenuStyle.leaderboardIcon}/>
              <Text style={MainMenuStyle.leaderboardText}>leaderboard</Text>
            </TouchableOpacity>
             {/*Should this ↑ touchableOpacity be its own component? */}


             <View style={MainMenuStyle.bottomArea}>
              <View style = {MainMenuStyle.copyrightArea}>
                <Text style={[MainMenuStyle.copyrightText, { color: "#F1C431" }]}>sound effects by:</Text>
                <Text style={[MainMenuStyle.copyrightText, { color: "#68CC73" }]}>music by:</Text>
                <Text style={[MainMenuStyle.copyrightText, { color: "#3998DB" }]}>Based on "colorblinder" by Daniel Gergely</Text>
              </View>
              <TouchableOpacity onPress={onToggleSound}>
                  <Image source={changeSoundIcon()} style={MainMenuStyle.soundIcon} />
              </TouchableOpacity>
             </View>

          </View>
  );
}

