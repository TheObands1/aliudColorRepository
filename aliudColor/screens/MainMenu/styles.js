import { StyleSheet } from "react-native";

export default StyleSheet.create({
 generalStyle: {
   flex: 1,
   backgroundColor: "#0a0a0a",
   justifyContent: "center",
   alignItems: "center",
 },
 playButtonArea: {
    flexDirection: 'row', 
    alignItems: 'center' ,
    marginTop: 80
 },
 playIcon: {
    height: 60,
    width: 60,
    marginRight: 15
  },
 playText: {
    textTransform: 'uppercase',
    fontSize: 45,
    color: "#ecf0f1",
    fontFamily: "dogbyte",
    marginTop: 5
  },
  highScoreArea:{
    flexDirection: 'row', 
    alignItems: 'center' ,
    marginTop: 20
  },
  highScoreText:{
    textTransform: 'capitalize',
    fontSize: 28.5,
    fontFamily: "dogbyte",
    color: "#ecf0f1",
    marginTop: 5
  },
  highScoreIcon:{
    height: 45,
    width: 45,
    marginRight: 12.5
},
leaderboardArea:{
    flexDirection: 'row', 
    alignItems: 'center' ,
    marginTop: 80
  },
  leaderboardText:{
    textTransform: 'capitalize',
    fontSize: 32,
    fontFamily: "dogbyte",
    color: "#ecf0f1",
    marginTop: 5
  },
  leaderboardIcon:{
    height: 52,
    width: 52,
    marginRight: 10
}
});