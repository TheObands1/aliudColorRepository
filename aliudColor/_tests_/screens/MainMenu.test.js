import React from 'react';
import renderer from 'react-test-renderer';
import MainMenu from "../../screens/MainMenu/index"
import { NavigationContext } from "@react-navigation/native"
import { render, fireEvent, waitFor } from '@testing-library/react-native';

describe('tests for the <MainMenu/>', () => {
    
    const navContext = {
        isFocused: () => true,
        addListener: jest.fn(() => jest.fn())
      }
    it('Renders the component correctly', () => {
        const navigate = jest.fn();
        const tree = renderer.create(   <NavigationContext.Provider value={navContext}>
                                            <MainMenu navigation={{ navigate }} />
                                        </NavigationContext.Provider>
        );
        expect(tree).toMatchSnapshot();
    });
    it('Navigates to instructions screen when button is pressed', () => {
        const navigate = jest.fn();
        const { getByText } = render( <NavigationContext.Provider value={navContext}>
                                                        <MainMenu navigation={{ navigate }} />
                                                    </NavigationContext.Provider>);
        fireEvent.press(getByText("instructions"));
        expect(navigate).toHaveBeenCalledWith('Instructions');
      });

  });

  

 