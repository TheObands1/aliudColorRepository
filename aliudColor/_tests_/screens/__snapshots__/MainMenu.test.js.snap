// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the <MainMenu/> Renders the component correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0a0a0a",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityLabel="aliudColor"
    accessible={true}
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "dogbyte",
          },
          Object {
            "color": "#ecf0f1",
            "fontSize": 55,
          },
        ]
      }
    >
      aliud
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "dogbyte",
          },
          Object {
            "color": "#E64C3C",
            "fontSize": 55,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      c
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "dogbyte",
          },
          Object {
            "color": "#E57E31",
            "fontSize": 55,
          },
        ]
      }
    >
      o
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "dogbyte",
          },
          Object {
            "color": "#F1C431",
            "fontSize": 55,
          },
        ]
      }
    >
      l
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "dogbyte",
          },
          Object {
            "color": "#68CC73",
            "fontSize": 55,
          },
        ]
      }
    >
      o
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "dogbyte",
          },
          Object {
            "color": "#3998DB",
            "fontSize": 55,
          },
        ]
      }
    >
      r
    </Text>
  </View>
  <View
    accessibilityHint="Press to play the game"
    accessibilityLabel="Play Button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 80,
        "opacity": 1,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 60,
          "marginRight": 15,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#ecf0f1",
          "fontFamily": "dogbyte",
          "fontSize": 45,
          "marginTop": 5,
          "textTransform": "uppercase",
        }
      }
    >
      play!
    </Text>
  </View>
  <View
    accessibilityLabel="High-Score: 0"
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 30,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 45,
          "marginRight": 12.5,
          "width": 45,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#ecf0f1",
          "fontFamily": "dogbyte",
          "fontSize": 28.5,
          "marginTop": 5,
          "textTransform": "capitalize",
        }
      }
    >
      high-score: 
      0
    </Text>
  </View>
  <View
    accessibilityHint="Press to go to the instructions page"
    accessibilityLabel="Instructions Button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#68CC73",
          "fontFamily": "dogbyte",
          "fontSize": 60,
          "marginRight": 10,
        }
      }
    >
      i
    </Text>
    <Text
      style={
        Object {
          "color": "#ecf0f1",
          "fontFamily": "dogbyte",
          "fontSize": 32,
          "marginTop": 5,
          "textTransform": "capitalize",
        }
      }
    >
      instructions
    </Text>
  </View>
  <View
    accessibilityHint="Press to toggle accesibility mode on"
    accessibilityLabel="Accessibility Button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ecf0f1",
          "fontFamily": "dogbyte",
          "fontSize": 30,
          "marginTop": 5,
          "textTransform": "capitalize",
        }
      }
    >
      accessibility mode: 
    </Text>
    <Image
      source={1}
      style={
        Object {
          "height": 52,
          "marginLeft": 10,
          "width": 52,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "bottom": 20,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "left": 15,
        "position": "absolute",
        "right": 15,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#ecf0f1",
              "fontFamily": "dogbyte",
              "fontSize": 13,
              "textTransform": "capitalize",
            },
            Object {
              "color": "#3998DB",
            },
          ]
        }
      >
        based on "colorblinder" by Daniel Gergely
      </Text>
    </View>
  </View>
</View>
`;
